### PR TITLE
Add network

### PR DESCRIPTION
**Pull request by berkbaski**

### CODE REVIEW

File: device/network/network-Android.js [hunk 1]
```diff
@@ -167,12 +168,35 @@ Network.createNotifier = function(params) {
...
 
+Network.isConnected = function () {
+    return new Promise((resolve, reject) => {
+        const noConnection = Network.ConnectionType === Network.ConnectionType.NONE;
+        if (noConnection) {
+            return reject();
+        }
+        const http = new Http();
+        http.request({
+            url: "https://www.google.com",
```
Comment (ozcanovunc) [inline device/network/network-Android.js:185]: I would also check for parameter, Google might be blocked for some private networks

Comment (ozcanovunc) [reply device/network/network-Android.js:185]: Network.isConnected('turkcell.com/something')

File: device/network/network-iOS.js [hunk 2]
```diff
@@ -109,7 +111,30 @@ Network.createNotifier = function(params) {
 
     //Android specs
     self.android = {};
-    self.android.isInitialStickyNotification = () => {};
+    self.android.isInitialStickyNotification = () => { };
+}
+
+Network.isConnected = function () {
```
Comment (ozcanovunc) [inline device/network/network-iOS.js:117]: Duplicate

**Review by ozcanovunc — CHANGES_REQUESTED** [review]

File: device/network/network-Android.js (hunk 1 above)
Comment (berkbaski) [reply device/network/network-Android.js]: I added the parameter for this. Could you check it again? @ozcanovunc

File: device/network/network-iOS.js (hunk 2 above)
Comment (berkbaski) [reply device/network/network-iOS.js]: I didn't understand which function is duplicated? @ozcanovunc

Comment (berkbaski) [reply device/network/network-iOS.js]: Alright, I understand now. I changed it, could you check it? @ozcanovunc

**Comment by ozcanovunc** [comment]
@berkbaski 💯

**Review by ozcanovunc — APPROVED** [review]

File: device/network/index.d.ts [hunk 3]
```diff
@@ -97,95 +97,114 @@ declare type NetworkNotifier = {
...
   static createNotifier(): NetworkNotifier;
+
+  /**
+   * @function
+   * @param checkUrl Used URL for check the internet
+   * @return {Promise<Object>} - Resolves if the internet connectivity is available,
+   * rejects o/w
+   * @example
+   * ```
+   * import network from '@smartface/extension-utils/lib/network';
```
Comment (alimoncul) [inline device/network/index.d.ts:197]: this path needs to updated right?

Comment (berkbaski) [reply device/network/index.d.ts]: Done 👍🏻

File: device/network/network-Android.js (hunk 1 above)
Comment (furkanarabaci) [reply device/network/network-Android.js]: We might actually default to ifconfig.co or something that wouldn't be blocked by anything.

**Review by furkanarabaci — APPROVED** [review]

**Comment by furkanarabaci** [comment]
Looks awesome.

**Pull request merged**